analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/SymbolicContextTest.cs

```diff
@@ -116,6 +117,18 @@ public void WithState_DifferentState_ReturnsNew()
             sut.WithState(newState).Should().NotBe(sut);
         }
 
+        [TestMethod]
+        public void ApplyOpposite_RespectsArgument()
+        {
+            BoolConstraint.True.ApplyOpposite(false).Should().Be(BoolConstraint.True);
+            BoolConstraint.True.ApplyOpposite(true).Should().Be(BoolConstraint.False);
+            BoolConstraint.False.ApplyOpposite(false).Should().Be(BoolConstraint.False);
+            BoolConstraint.False.ApplyOpposite(true).Should().Be(BoolConstraint.True);
+            // Special constraint behavior
+            ObjectConstraint.Null.ApplyOpposite(true).Should().Be(ObjectConstraint.NotNull);
+            ObjectConstraint.NotNull.ApplyOpposite(true).Should().BeNull("because NotNull can be Null or any other NotNull");
```

Comment:
NIT: I think the resulting assertion message would read "because because". 
"the opposite of NotNull can be Null or any other NotNull"